File: python3/leetcode-283.py
# python3
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = 0
        len_nums = len(nums)
        for i in range(len_nums):
            if nums[i] != 0:
                nums[n] = nums[i]
                n += 1
        while n < len_nums:
            nums[n] = 0
            n += 1

File: go/leetcode-283.go
package main

import "fmt"

//283. 移动零
func moveZeroes1(nums []int) {
	var n int
	for i := 0; i < len(nums); i++ {
		if nums[i] != 0 {
			nums[n] = nums[i]
			n++
		}

	}
	for {
		if n < len(nums) {
			nums[n] = 0
			n++
		} else {
			break
		}

	}
}
func main() {
	nums := []int{0, 1, 0, 3, 12}
	moveZeroes1(nums)
	fmt.Println(nums)
}
